proposals/web-api-add-edit-with-media.md

```diff
@@ -0,0 +1,618 @@
+    - [Upload the Media byte array](#upload-the-media-byte-array)
+    - [Successful Media Upload](#successful-media-upload)
+    - [Media Record created but no byte stream provided](#media-record-created-but-no-byte-stream-provided)
+    - [Error Uploading Media](#error-uploading-media)
+    - [Media uploaded but byte stream rejected](#media-uploaded-but-byte-stream-rejected)
+    - [Attempting to Update a Write-Once Data Provider](#attempting-to-update-a-write-once-data-provider)
+- [Section 3: Certification](#section-3-certification)
+- [Section 4: Contributors](#section-4-contributors)
+- [Section 5: References](#section-5-references)
+- [Section 6: Appendices](#section-6-appendices)
+- [Section 7: License](#section-7-license)
+
+<br /><br />
+
+
+# Section 1: Introduction
+Solutions using the RESO Web API need a way to add or edit media data such as photos, documents and videos so that Web API clients can provide the media binary data consistently to the servers regardless of the Web API provider.  This proposal states a solution that leveraging the existing Add/Edit process and the OData binary data specifications.
+
+<br /><br />
+
+# Summary of Changes
+
+* Introduces a specification for the addition and update of Media binary byte stream data to the transport specification
+* Changes `MediaStatus` into a "Open With Enumerations" field
+* Adds a `MediaStatusDescription` field to the `Media` resource
+* Add a `hasStreamYN` field to the `Model` resource
+
+<br /><br />
+
+# Section 2: Specification
+
+This specification relies heavily on the [OData streams](https://docs.oasis-open.org/odata/odata-json-format/v4.0/os/odata-json-format-v4.0-os.html) specification for the transport interaction with the server.
+
+While OData streams works for the communication of the data, there are some media operations that cannot be preformed in real 
+time and are deferred post-processing that could prevent the successful publishing of the media.  This could include format
+validation, transcoding (resizing, reformating) for different displays/devices, compliance validation, copyright validation and/or content delivery distribution to name a few identified scenarios.  
+As a result of this need to allow for background operations to occur after the media has been received, a method to communicate the progression of the media is required.  The existing field `MediaStatus` will be used to comminucate the status and `MediaStatusDescription` will be introduced to provide contextual information when available.  This will assist clients in know why media has been rejected without having to contact the server provider outside of the transport.
+
+## Data Representation
+
+### Data Structure
+These are the required fields to represent the upload process for media.  The data dictionary change set can be found [here](https://docs.google.com/spreadsheets/d/1QGAe3hFvKHu9fMVzlGbI4gIuiSLjdTHO7leGK47bP88/edit?usp=sharing):
```

Comment:
This shouldn't be linked to in external sheet, but rather outlined in this section (which I think they are).